suprb/suprb.py

```diff
@@ -147,15 +147,27 @@ def fit(self, X: np.ndarray, y: np.ndarray, cleanup=False):
 
         # Fill population before first step
         if self.n_initial_rules > 0:
-            self._discover_rules(X, y, self.n_initial_rules)
+            try:
+                self._discover_rules(X, y, self.n_initial_rules)
+            except:
+                warnings.warn("Error in rule discovery first iteration")
+                return self
```

Comment:
what is the fitness of the returned model? it should be horrificly bad

Reply:
Changed it, so in predict we get a very bad MSE (almost 1, something like 0.99999) when we enter an error case